src/main/kotlin/LinkedList.kt

```diff
@@ -10,18 +10,60 @@ class LinkedList<T> {
     
     fun count(): Int = size
        
-    fun insert(index: Int, obj: T) {
-        TODO("Not yet implemented")
-    }
-    fun removeAt(index: Int) {
-        TODO("Not yet implemented")
+    private fun getNode(index: Int): Node<T> {
+        var current = head
+        var currentIndex = 0
+        while (current != null && currentIndex < index) {
+            current = current.nextNode
+            currentIndex++
+        }
+        return current!!
     }
+    
     operator fun get(index: Int): T {
-        TODO("Not yet implemented")
+        require(index in 0 until size-1)
+        return getNode(index).value
     }
+    
     operator fun set(index: Int, obj: T) {
         TODO("Not yet implemented")
     }
+    
+    fun insertFirst(value: T) {
+        head = Node(value = value, nextNode = head)
+        if (last == null) {
+            last = head
+        }
+        size++
+    }
+
+    fun insertLast(value: T) {
+        if (size==0) {
+            insertFirst(value)
+        }
+        else {
+            last?.nextNode = Node(value = value)
+            last= last?.nextNode
+            size++
+        }
+    }
+    
+    fun insert(index: Int, value: T) {
```

Comment:
Так или иначе в начале нужно сделать проверку, что индекс вменяемый:
```suggestion
    fun insert(index: Int, value: T) {
        require(index in 0 .. size)
```

Reply:
добавил